src/api/catalog-search/util.ts

```diff
@@ -11,13 +12,31 @@ type FilterFunctionBuilder<T> = (
   filter: T
 ) => undefined | ((pkg: ExtendedCatalogPackage) => boolean);
 
+const getVersionSort =
+  (isAscending: boolean): SortFunction =>
+  (p1, p2) =>
+    compare(p1.version, p2.version) * (isAscending ? 1 : -1);
+
+const getStrSort = (isAscending: boolean): SortFunction => {
+  return (p1, p2) => {
+    if (p1.name === p2.name) {
+      // Sort by versions if the names are identical.
+      return getVersionSort(isAscending)(p1, p2);
+    }
+    return p1.name.localeCompare(p2.name) * (isAscending ? 1 : -1);
+  };
+};
+
 const getDateSort =
   (isAscending: boolean): SortFunction =>
   (p1, p2) => {
-    const d1 = new Date(p1.metadata.date);
-    const d2 = new Date(p2.metadata.date);
+    // Sorting only by the DATE portion of the timestamp. ISOString is YYYY-MM-DD'T'HH:mm:ss.SSSZ
+    const d1 = new Date(p1.metadata.date).toISOString().split("T")[0];
+    const d2 = new Date(p2.metadata.date).toISOString().split("T")[0];
+
+    // If they're equal, fall back to alphanumerical ordering.
     if (d1 === d2) {
-      return 0;
+      return getStrSort(!isAscending)(p1, p2);
```

Comment:
Thought I left a comment here, but I strongly believe we should avoid sorting just by date - this breaks the newest / oldest flow quite a bit because as your above comment said, all results will just be alphabetical for the same day and it's going to be very odd from a user perspective